var/spack/repos/builtin/packages/bambam/package.py

```diff
@@ -0,0 +1,43 @@
+# Copyright 2013-2018 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Bambam(MakefilePackage):
+    """Bambam is a tool used to facilitate NGS analysis."""
+
+    homepage = "http://udall-lab.byu.edu/Research/Software/BamBam"
+    url      = "https://downloads.sourceforge.net/project/bambam/bambam-1.4.tgz"
+
+    version('1.4', sha256='a9b178251d771aafb8c676d30a9af88ea2fc679c2a5672b515f86be0e69238f1')
+    version('1.3', sha256='de91af03c1c09921a3a7f816b4df0564213e57c1a6024f2bcd63c0e8f0733a50', preferred=True)
+
+    depends_on('perl', type=('build', 'run'))
+    depends_on('samtools')
+    depends_on('bamtools@2.3.0')
+    depends_on('htslib')
+    depends_on('zlib')
+
+    def edit(self, spec, prefix):
+        makefile = FileFilter('makefile')
+        makefile.filter('INC = *', 'INC = -I%s ' %
+                        self.spec['bamtools'].prefix.include)
+
+    def install(self, spec, prefix):
+        install_tree('bin', prefix.bin)
+        install_tree('lib', prefix.lib)
+        install_tree('scripts', prefix.scripts)
+
+    def setup_environment(self, spack_env, run_env):
+        spack_env.prepend_path('LIBRARY_PATH',
```

Comment:
this should automatically be done if you declare samtools to be a run dependency

Reply:
or at least it should be samtools' job to set up this environment

Reply:
run dependencies are only added to `PATH`, not `LIBRARY_PATH`. I'm surprised that this is necessary given the compiler wrappers.

Reply:
yeah, I was surprised as well, but I could not get it to work without it.  Here's what ldd looks like on polyCat without the module loaded (setting the LIBRARY_PATH)):
`linux-vdso.so.1 =>  (0x00007ffffead1000)
	libbamtools.so.2.3.0 => not found
	libhts.so.2 => /opt/rit/spack-app/linux-rhel7-x86_64/gcc-4.8.5/htslib-1.9-xjiydchack6jgcsr6pgcjewekddibuvl/lib/libhts.so.2 (0x00007f941759a000)
	libz.so.1 => /opt/rit/spack-app/linux-rhel7-x86_64/gcc-4.8.5/zlib-1.2.11-vhzh5cfaki5lx5sjuth5iuojq5azdkbd/lib/libz.so.1 (0x00007f9417383000)
	libstdc++.so.6 => /lib64/libstdc++.so.6 (0x00007f941707c000)
	libm.so.6 => /lib64/libm.so.6 (0x00007f9416d7a000)
	libgcc_s.so.1 => /lib64/libgcc_s.so.1 (0x00007f9416b64000)
	libpthread.so.0 => /lib64/libpthread.so.0 (0x00007f9416948000)
	libc.so.6 => /lib64/libc.so.6 (0x00007f941657b000)
	liblzma.so.5 => /opt/rit/spack-app/linux-rhel7-x86_64/gcc-4.8.5/xz-5.2.4-rh7gav63dk5xbjp2hq4rtp4wmuwk7slb/lib/liblzma.so.5 (0x00007f9416356000)
	libbz2.so.1.0 => /opt/rit/spack-app/linux-rhel7-x86_64/gcc-4.8.5/bzip2-1.0.6-qrd3gzlkqpof6jthkmt2vejxvmbd3x3s/lib/libbz2.so.1.0 (0x00007f9416146000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f9417849000)
`